File: SwiftOSXProofOfConcept/data2yaml/data/random.lua
sptbl["random"] = {

    files = { 
        module = "random.c",
        header = "random.h",
        example = "ex_random.c",
    },
    
    func = {
        create = "sp_random_create",
        destroy = "sp_random_destroy",
        init = "sp_random_init",
        compute = "sp_random_compute",
    },
    
    params = {
        optional = {
            {
                name = "min",
                type = "SPFLOAT",
                description = "Minimum value.",
                default = -0.2
            },
            {
                name = "max",
                type = "SPFLOAT",
                description ="Maximum value.",
                default = 0.2
            },
        }
    },
    
    modtype = "module",
    
    description = [[Generate random values within a range.]], 
    
    ninputs = 0,
    noutputs = 1,
    
    inputs = { 
        {
            name = "dummmy",
            description = "This doesn't do nuthin."
        },
    },
    
    outputs = {
        {
            name = "out",
            description = "Signal output."
        },
    }

}

File: AudioKit/Common/Internals/Soundpipe/modules/random.c
#include <stdlib.h>
#include "soundpipe.h"

int sp_random_create(sp_random **p)
{
    *p = malloc(sizeof(sp_random));
    return SP_OK;
}

int sp_random_destroy(sp_random **p)
{
    free(*p);
    return SP_OK;
}

int sp_random_init(sp_data *sp, sp_random *p)
{
    p->min = -1;
    p->max = 1;
    return SP_OK;
}

int sp_random_compute(sp_data *sp, sp_random *p, SPFLOAT *in, SPFLOAT *out)
{
    /* Send the signal's input to the output */
    SPFLOAT rnd = ((sp_rand(sp) % RAND_MAX) / (RAND_MAX * 1.0));
    rnd *= (p->max - p->min);
    rnd += p->min;
    *out = rnd;
    return SP_OK;
}
